### PR TITLE
[No Ticket] Refactor query building

### PR DESCRIPTION
**Pull request by mschuwalow**
Author Todo List:

- [x] Add/adjust tests (if applicable)
- [x] Build in CI passes
- [x] Latest master revision is merged into the branch
- [x] Self-Review
- [x] Set `Ready For Review` status

### CODE REVIEW

File: src/idex.ts [hunk 1]
```diff
@@ -48,25 +47,24 @@ export class IdentityResolver {
...
+      .addOptionalParam('duid', nonNullConfig.peopleVerifiedId)
+      .addOptionalParam('us_privacy', nonNullConfig.usPrivacyString)
+      .addOptionalParam('gdpr', onNonNull(nonNullConfig.gdprApplies, v => v ? 1 : 0))
+      .addOptionalParam('n3pc', nonNullConfig.privacyMode ? 1 : undefined)
+      .addOptionalParam('gdpr_consent', nonNullConfig.gdprConsent)
+      .addOptionalParam('did', nonNullConfig.distributorId)
+      .addOptionalParam('gpp_s', nonNullConfig.gppString)
+      .addOptionalParam('gpp_as', nonNullConfig.gppApplicableSections?.join(','))
+      .addOptionalParam('cd', nonNullConfig.cookieDomain)
+      .addOptionalParam('ic', encodeIdCookie(nonNullConfig.resolvedIdCookie))
```
Comment (mschuwalow) [inline src/idex.ts:60]: @3link should we send empty ones (failed resolution) to idex as well or is live-pixel enough?

File: src/idex.ts [hunk 2]
```diff
@@ -21,7 +20,7 @@ export class IdentityResolver {
   url: string
   timeout: number
   requestedAttributes: string[]
-  tuples: [string, string][]
+  query: QueryBuilder // ensure to copy this before mutating
```
Comment (3link) [inline src/idex.ts:23]: Find the comment really helpful. But maybe provide more details here? Something like "Be careful, this object is mutable. In cases where temporary changes are needed - e.g. adding parameters that are only valid for one single call - ensure to copy it".

**Review by 3link — APPROVED** [review]
Please see comments.

File: src/idex.ts (hunk 1 above)
Comment (3link) [inline src/idex.ts:60]:
```suggestion
      .addOptionalParam('ic', encodeIdCookie(nonNullConfig.resolvedIdCookie), { stripEmpty: false })
```

File: src/utils/query.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { isArray, nonNull, onNonNull } from 'live-connect-common'
+import { md5 } from 'tiny-hashes/dist'
+
+export type ParamValue = string | number | boolean
+export type ParamOptions = {
+  stripEmpty?: boolean
+  prepend?: boolean
+}
+
+export class QueryBuilder {
```
[the remark this answers is not in the document]
Comment (mschuwalow) [reply src/utils/query.ts:10]: it's tested transitively through the existing flows. But sure, will add a dedicated suite.

**Review by wi101 — APPROVED** [review]

**Review by peixunzhang — APPROVED** [review]

**Pull request merged**